iib/workers/tasks/build_merge_index_image.py

```diff
@@ -285,3 +288,58 @@ def handle_merge_request(
     set_request_state(
         request_id, 'complete', 'The index image was successfully cleaned and updated.'
     )
+
+
+def is_bundle_version_valid(bundle_path, valid_ocp_version):
+    """
+    Check if the version label of the bundle satisfies the index ocp_version.
+
+    :param str bundle_path: pull specification of the bundle to be validated.
+    :param str valid_ocp_version: the index ocp version against which the bundles will be validated.
+    :return: a boolean indicating if the bundle_path satisfies the index ocp_version
+    :rtype: bool
+
+           |  "v4.5"   |   "=v4.6"    | "v4.5-v4.7" | "v4.5,v4.6"
+    ---------------------------------------------------------------
+    v4.5   | included  | NOT included |  included   |  included
+    ---------------------------------------------------------------
+    v4.6   | included  |   included   |  included   |  included
+    ---------------------------------------------------------------
+    v4.7   | included  | NOT included |  included   |  included
+    """
+    try:
+        float_valid_ocp_version = float(valid_ocp_version.replace('v', ''))
+    except ValueError:
+        raise IIBError(f'Invalid ocp_version "{valid_ocp_version}" specified for target_index')
+    try:
+        bundle_version_label = get_image_label(bundle_path, 'com.redhat.openshift.versions')
+        bundle_version = bundle_version_label.replace('v', '')
+        if bundle_version_label.startswith('='):
+            if float(bundle_version.strip('=')) == float_valid_ocp_version:
+                return True
+        elif '-' in bundle_version_label:
+            min_version, max_version = [float(version) for version in bundle_version.split('-')]
+            if min_version <= float_valid_ocp_version <= max_version:
+                return True
+        elif "," in bundle_version_label:
+            versions = [float(version) for version in bundle_version.split(",")]
+            # This means the version is something like v4.6, v4.5 which is not valid
+            if versions != sorted(versions):
+                log.warning(
```

Comment:
Just raise a `ValueError` so it goes through the same path as the other invalid cases.

Reply:
+1

Reply:
Also added a test to cover that.